diff -up gambas-3.19.5/gb.pdf/src/CPdfDocument.cpp.omv~ gambas-3.19.5/gb.pdf/src/CPdfDocument.cpp
--- gambas-3.19.5/gb.pdf/src/CPdfDocument.cpp.omv~	2024-10-21 14:52:14.557973950 +0200
+++ gambas-3.19.5/gb.pdf/src/CPdfDocument.cpp	2024-10-21 14:52:28.468124415 +0200
@@ -1037,7 +1037,7 @@ BEGIN_METHOD(PDFPAGE_select, GB_INTEGER
 	h = VARGOPT(H, (int32_t)THIS->page->getMediaHeight());
 
 	dev = new TextOutputDev (NULL, true, 0, false, false);
-	gfx = THIS->page->createGfx(dev,72.0,72.0,0,false,true,-1, -1, -1, -1, false, NULL, NULL);
+	gfx = THIS->page->createGfx(dev,72.0,72.0,0,false,true,-1, -1, -1, -1, nullptr, NULL, NULL);
 
 	THIS->page->display(gfx);
 	dev->endPage();
