diff -up main/configure.ac.noliconv main/configure.ac
--- gambas3-3.2.0/main/configure.ac.noliconv	
+++ gambas3-3.2.0/main/configure.ac	
@@ -27,8 +27,8 @@ dnl ---- Check for charset conversion li
 GB_COMPONENT(
   conv, CONV, libiconv, [],
   [GB_FIND(iconv.h, /opt/local /usr/local /usr, include)],
-  [GB_FIND(libiconv.$SHLIBEXT, /opt/local /usr/local /usr, lib)],
-  [-liconv],
+  [],
+  [],
   [],
   [This library may be located inside the system C library, so you can ignore that warning...])
 
