diff -up gambas-3.18.3/gb.qt5/src/main.cpp.omv~ gambas-3.18.3/gb.qt5/src/main.cpp
--- gambas-3.18.3/gb.qt5/src/main.cpp.omv~	2023-10-06 18:59:27.666478107 +0200
+++ gambas-3.18.3/gb.qt5/src/main.cpp	2023-10-06 19:01:09.894425782 +0200
@@ -733,7 +733,7 @@ static void hook_main(int *argc, char **
 	
 	platform = qApp->platformName();
 
-	if (platform == "wayland")
+	if (platform.startsWith("wayland"))
 	{
 		comp = "gb.qt5.wayland";
 		MAIN_platform = "wayland";
