diff -up gambas-3.19.1/main/share/gambas.h.omv~ gambas-3.19.1/main/share/gambas.h
--- gambas-3.19.1/main/share/gambas.h.omv~	2024-05-04 03:08:33.986115907 +0200
+++ gambas-3.19.1/main/share/gambas.h	2024-05-04 03:12:27.703670630 +0200
@@ -1304,6 +1304,10 @@ typedef
 
 #include <string.h>
 #include <ctype.h>
+#ifdef __cplusplus
+#include <format> // must be included before redefinition of toupper...
+#include <chrono> // must be included before redefinition of toupper...
+#endif
 
 #ifdef tolower
 #undef tolower
