diff -up gambas-3.20.0/gb.pdf/src/CPdfDocument.cpp.fix gambas-3.20.0/gb.pdf/src/CPdfDocument.cpp
--- gambas-3.20.0/gb.pdf/src/CPdfDocument.cpp.fix	2025-01-08 17:00:44.000000000 -0500
+++ gambas-3.20.0/gb.pdf/src/CPdfDocument.cpp	2025-01-17 11:37:52.831010018 -0500
@@ -1027,8 +1027,8 @@ END_PROPERTY
 BEGIN_METHOD(PDFPAGE_select, GB_INTEGER X; GB_INTEGER Y; GB_INTEGER W; GB_INTEGER H)
 
 	TextOutputDev *dev;
-	GooString *str;
-	Gfx *gfx;
+	GooString str;
+	std::unique_ptr<Gfx> gfx;
 	int32_t x,y,w,h;
 
 	x = VARGOPT(X, 0);
@@ -1043,22 +1043,14 @@ BEGIN_METHOD(PDFPAGE_select, GB_INTEGER
 	gfx = THIS->page->createGfx(dev,72.0,72.0,0,false,true,-1, -1, -1, -1, false, NULL, NULL);
 #endif
 
-	THIS->page->display(gfx);
+	THIS->page->display(gfx.get());
 	dev->endPage();
 
 	str=dev->getText((double)x,(double)y,(double)(w+x),(double)(h+y));
 
-	delete gfx;
 	delete dev;
 
-	if (!str)
-	{
-		GB.ReturnNewZeroString("");
-		return;
-	}
-	
-	GB.ReturnNewString(str->getCString(),str->getLength());	
-	delete str;
+	GB.ReturnNewString(str.getCString(),str.getLength());
 
 END_METHOD
 
