diff -up gambas-3.18.0/gb.pdf/src/CPdfDocument.cpp.1~ gambas-3.18.0/gb.pdf/src/CPdfDocument.cpp
--- gambas-3.18.0/gb.pdf/src/CPdfDocument.cpp.1~	2023-01-12 23:31:22.000000000 +0100
+++ gambas-3.18.0/gb.pdf/src/CPdfDocument.cpp	2023-02-03 00:08:46.988368167 +0100
@@ -1038,11 +1038,7 @@ BEGIN_PROPERTY (PDFPAGELINKS_count)
 	if (!THIS->links) { GB.ReturnInteger(0); return; }
 
 #if POPPLER_VERSION_22_06_0
-	int numlinks = 0;
-	for (AnnotLink *link : THIS->links->getLinks()) {
-		numlinks++;
-	}
-	GB.ReturnInteger(numlinks);
+	GB.ReturnInteger(THIS->links->getLinks().size());
 #else
 	GB.ReturnInteger(THIS->links->getNumLinks());
 #endif
@@ -1078,14 +1074,14 @@ BEGIN_METHOD (PDFPAGELINKS_get,GB_INTEGE
 		if (VARG(ind)<0) pok=false;
 		else
 		{
-			if (VARG(ind)>=THIS->links->getNumLinks()) pok=false;
+			if (VARG(ind)>=THIS->links->getLinks().size()) pok=false;
 		}
 	}
 
 	if (!pok) { GB.Error("Out of bounds"); return; }
 
 	THIS->lcurrent=VARG(ind);
-	THIS->action=THIS->links->getLink(THIS->lcurrent)->getAction();
+	THIS->action=THIS->links->getLinks().at(THIS->lcurrent)->getAction();
 
 	RETURN_SELF();
 
