--- gambas3-3.9.2/gb.sdl2/configure.ac.omv~	2017-01-23 02:48:11.753920392 +0100
+++ gambas3-3.9.2/gb.sdl2/configure.ac	2017-01-23 02:48:28.578017862 +0100
@@ -9,12 +9,12 @@ AC_PROG_LIBTOOL
 
 GB_COMPONENT_PKG_CONFIG(
    sdl2, SDL2, gb.sdl2, [src],
-   sdl2 ">= 2.0.2" SDL2_image ">= 2.0.0" SDL2_ttf ">= 2.0.12"
+   sdl2 SDL2_image SDL2_ttf
 )
 
 GB_COMPONENT_PKG_CONFIG(
    sdl2audio, SDL2AUDIO, gb.sdl2.audio, [audio],
-   sdl2 ">= 2.0.2" SDL2_mixer ">= 2.0.0"
+   sdl2 SDL2_mixer
 )
 
 AC_OUTPUT( \
