diff -up gambas-3.18.4/main/lib/gui/gb_gui_test_temp.h.omv~ gambas-3.18.4/main/lib/gui/gb_gui_test_temp.h
--- gambas-3.18.4/main/lib/gui/gb_gui_test_temp.h.omv~	2023-10-06 20:09:43.971794560 +0200
+++ gambas-3.18.4/main/lib/gui/gb_gui_test_temp.h	2023-10-06 20:11:05.598495105 +0200
@@ -73,11 +73,14 @@ static int GUI_should_use()
 			{
 				if (strcmp(env, "4") == 0)
 					return USE_GB_QT4;
-				else if (strcmp(env, "5") == 0)
+				else
 					return USE_GB_QT5;
 			}
 		}
-	}
+	} else if(env && !strncasecmp(env, "LXQt", 4))
+		return USE_GB_QT5;
+	else if(env && (!strncasecmp(env, "gnome", 5) || !strncasecmp(env, "mate", 4) || !strncasecmp(env, "cinnamon", 8) || !strncasecmp(env, "xfce", 4)))
+		return USE_GB_GTK3;
 
 	return USE_NOTHING;	
 }
diff -up gambas-3.18.4/main/lib/gui/main.c.omv~ gambas-3.18.4/main/lib/gui/main.c
--- gambas-3.18.4/main/lib/gui/main.c.omv~	2023-10-06 20:11:11.257544032 +0200
+++ gambas-3.18.4/main/lib/gui/main.c	2023-10-06 20:12:02.273987042 +0200
@@ -53,9 +53,9 @@ const char *get_name(int use)
 	switch (use)
 	{
 		case USE_GB_QT4: return "gb.qt4";
-		case USE_GB_QT5: return "gb.qt5";
+		case USE_GB_GTK: return "gb.gtk";
 		case USE_GB_GTK3: return "gb.gtk3";
-		default: return "gb.gtk";
+		default: return "gb.qt5";
 	}
 }
 
