diff -up gambas-3.18.0/gb.pdf/configure.ac.omv~ gambas-3.18.0/gb.pdf/configure.ac
--- gambas-3.18.0/gb.pdf/configure.ac.omv~	2023-02-03 01:47:38.384628112 +0100
+++ gambas-3.18.0/gb.pdf/configure.ac	2023-02-03 01:47:51.474746450 +0100
@@ -10,7 +10,7 @@ PKG_PROG_PKG_CONFIG
 
 GB_COMPONENT_PKG_CONFIG(
   poppler, POPPLER, gb.pdf, [src],
-  'poppler >= 0.20' 'poppler < 23.0'
+  'poppler >= 0.20' 'poppler < 30.0'
 )
 
 if test "$have_poppler" = "yes"; then
